py/fiberassign/stucksky.py

```diff
@@ -0,0 +1,189 @@
+
+    try:
+        skybricks = Skybricks()
+    except:
+        log.warning('Environment variable SKYBRICKS_DIR is not set; not looking up whether '
+                    'stuck positioners land on good sky')
+        return
+
+    stuck_sky = dict()
+    for tile_id, tile_ra, tile_dec, tile_theta in zip(
+            tiles.id, tiles.ra, tiles.dec, tiles.obstheta):
+        stuck_sky[tile_id] = dict()
+
+        # Stuck locations and their angles
+        stuck_loc = [loc for loc in hw.locations
+                     if ((hw.state[loc] & FIBER_STATE_STUCK != 0) and
+                         (hw.state[loc] & FIBER_STATE_BROKEN == 0) and
+                         (hw.loc_device_type[loc] == 'POS'))]
+        stuck_theta = [hw.loc_theta_pos[x] for x in stuck_loc]
+        stuck_phi   = [hw.loc_phi_pos  [x] for x in stuck_loc]
+
+        # Convert positioner angle orientations to curved focal surface X / Y (not CS5)
+        # Note:  we could add some methods to the python bindings to vectorize this or make it less clunky...
+        stuck_x = np.zeros(len(stuck_loc))
+        stuck_y = np.zeros(len(stuck_loc))
+        for iloc, (loc, theta, phi) in enumerate(zip(stuck_loc, stuck_theta, stuck_phi)):
+            loc_x, loc_y = hw.thetaphi_to_xy(
+                hw.loc_pos_curved_mm[loc],
+                theta,
+                phi,
+                hw.loc_theta_arm[loc],
+                hw.loc_phi_arm[loc],
+                hw.loc_theta_offset[loc],
+                hw.loc_phi_offset[loc],
+                hw.loc_theta_min[loc],
+                hw.loc_phi_min[loc],
+                hw.loc_theta_max[loc],
+                hw.loc_phi_max[loc]
+            )
+            stuck_x[iloc] = loc_x
+            stuck_y[iloc] = loc_y
+        
+        loc_radec = hw.xy2radec_multi(
+            tile_ra,
+            tile_dec,
+            tile_theta,
+            stuck_x,
+            stuck_y,
+            False,
+            0
+        )
+        loc_ra  = np.array([r for r,d in loc_radec])
+        loc_dec = np.array([d for r,d in loc_radec])
+
+        good_sky = skybricks.lookup_tile(tile_ra, tile_dec, hw.focalplane_radius_deg,
+                                         loc_ra, loc_dec)
+        log.debug('%i of %i stuck positioners land on good sky' % (np.sum(good_sky), len(good_sky)))
+
+        for loc,good in zip(stuck_loc, good_sky):
+            stuck_sky[tile_id][loc] = good
+
+    return stuck_sky
+
+def _radec2kd(ra, dec):
+    from scipy.spatial import KDTree
+    xyz = _radec2xyz(ra, dec)
+    return KDTree(xyz)
+
+def _radec2xyz(ra, dec):
+    rr = np.deg2rad(ra)
+    dd = np.deg2rad(dec)
+    xyz = np.zeros((len(rr),3))
+    xyz[:,0] = np.cos(rr) * np.cos(dd)
+    xyz[:,1] = np.sin(rr) * np.cos(dd)
+    xyz[:,2] = np.sin(dd)
+    return xyz
+            
+if __name__ == '__main__':
+    from fiberassign.tiles import load_tiles
+    from fiberassign.hardware import load_hardware
+
+    rundate = '2021-04-22T00:00:00'
```

Comment:
This was just for debugging the source code by importing the file directly, correct?

Reply:
Yep; removed!